core/src/worker/workflow/mod.rs

```diff
@@ -869,6 +874,18 @@ enum ActivationCompleteOutcome {
     ReportWFTFail(FailedActivationWFTReport),
     /// There's nothing to do right now. EX: The workflow needs to keep replaying.
     DoNothing,
+    /// The workflow task failed, but we shouldn't report it. EX: We have failed 2 or more attempts
+    /// in a row.
+    WFTFailedDontReport,
+}
+/// Did we report, or not, completion of a WFT to server?
+#[derive(Debug)]
+enum WFTReportStatus {
+    Reported,
+    NotReported,
+    /// We didn't report, but we want to clear the outstanding workflow task anyway. See
```

Comment:
I would add a comment to `NotReported`. Something like "we didn't report and the worker needs to continue working on this WFT task". Otherwise, the difference between `NotReported` and `DropWFT` is not very clear.